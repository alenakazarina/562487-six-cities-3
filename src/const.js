export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

export const OfferType = {
  house: `House`,
  room: `Private room`,
  apartment: `Apartment`,
  hotel: `Hotel`
};

export const AppRoute = {
  ROOT: `/`,
  OFFER: `/offer/:id`,
  FAVORITES: `/favorites`,
  LOGIN: `/login`
};

export const RATINGS = [
  {
    value: 5,
    title: `perfect`
  },
  {
    value: 4,
    title: `good`
  },
  {
    value: 3,
    title: `not bad`
  },
  {
    value: 2,
    title: `badly`
  },
  {
    value: 1,
    title: `terribly`
  }
];

export const MONTHS = [
  `January`, `February`, `March`, `April`,
  `May`, `June`, `July`, `August`,
  `September`, `October`, `November`, `December`
];

export const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const InitialState = {
  USER: {
    authStatus: `NO_AUTH`,
    user: {
      id: -1,
      email: ``,
      name: ``,
      isPro: false,
      avatarUrl: ``
    }
  },
  FAVORITES: {
    favorites: []
  },
  OFFER: {
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  OFFERS: {
    offers: [],
    activeCity: ``
  },
  ERRORS: {
    errorStatus: null
  }
};

