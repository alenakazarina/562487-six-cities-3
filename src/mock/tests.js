export const locations = [
  {
    title: `Paris`,
    isActive: true
  },
  {
    title: `Cologne`,
    isActive: false
  },
  {
    title: `Brussels`,
    isActive: false
  },
  {
    title: `Amsterdam`,
    isActive: false
  },
  {
    title: `Hamburg`,
    isActive: false
  },
  {
    title: `Dusseldorf`,
    isActive: false
  }
];

export const offers = [
  {
    id: `01`,
    title: `Beautiful & luxurious apartment at great location`
  },
  {
    id: `02`,
    title: `Wood and stone place`
  },
  {
    id: `03`,
    title: `Canal View Prinsengracht`
  },
  {
    id: `04`,
    title: `Nice, cozy, warm big bed apartment`
  },
  {
    id: `05`,
    title: `Cozy rooms in the city center`
  }
];

export const title = `Canal View Prinsengracht`;

export const onTitleClick = () => {};
