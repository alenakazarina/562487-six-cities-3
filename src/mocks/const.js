export const IMAGE_PATH = `./img/`;

const LOCATIONS_ZOOM = 13;

export const MONTHS = [
  `January`, `February`, `March`, `April`, `May`, `June`, `July`,
  `August`, `September`, `October`, `November`, `December`
];

const OfferTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

export const OffersData = {
  maxRating: 5,
  types: Object.values(OfferTypes),
  titles: [
    `Canal View room`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`,
    `Beautiful & luxurious apartment at great location`,
    `Cozy rooms in the city center`
  ],
  images: [
    `apartment-01.jpg`,
    `apartment-02.jpg`,
    `apartment-03.jpg`,
    `apartment-04.jpg`,
    `apartment-05.jpg`,
  ],
  amenities: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Cabel TV`, `Fridge`],
  description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
};

export const UserData = {
  names: [`Douglas`, `James`, `Matthew`, `John`, `Leo`, `Robert`, `Dirk`],
  avatarUrls: [
    `user-1.jpg`,
    `user-2.jpg`,
    `user-3.jpg`,
    `user-4.jpg`,
    `user-5.jpg`,
  ],
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
};

export const LOCATIONS = [
  {
    name: `Paris`,
    location: {
      latitude: 48.856663,
      longitude: 2.351556,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Cologne`,
    location: {
      latitude: 50.930779,
      longitude: 6.938399,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Brussels`,
    location: {
      latitude: 50.851309,
      longitude: 4.351718,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Amsterdam`,
    location: {
      latitude: 52.373057,
      longitude: 4.892557,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Hamburg`,
    location: {
      latitude: 53.552645,
      longitude: 9.966287,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Dusseldorf`,
    location: {
      latitude: 51.230569,
      longitude: 6.787428,
      zoom: LOCATIONS_ZOOM
    }
  }];
