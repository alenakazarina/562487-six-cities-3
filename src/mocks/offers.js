import {OffersData} from './const';
import {
  generateId,
  getImage,
  getRandomBoolean,
  getRandomNumber,
  getRandomFromArray,
} from './utils';

const generateOffer = () => {
  return {
    id: generateId(),
    previewImage: getImage(OffersData.images),
    title: getRandomFromArray(OffersData.titles),
    isFavorite: getRandomBoolean(),
    isPremium: getRandomBoolean(),
    rating: getRandomNumber({min: 0, max: OffersData.maxRating}),
    type: getRandomFromArray(OffersData.types),
    price: getRandomNumber({min: 0, max: 1000})
  };
};

export const generateOffers = (count = 3) => {
  return new Array(count).fill(``).map(generateOffer);
};

export const offers = [
  {
    city: `Paris`,
    offers: generateOffers(4)
  },
  {
    city: `Cologne`,
    offers: generateOffers(5)
  },
  {
    city: `Brussels`,
    offers: generateOffers(4)
  },
  {
    city: `Amsterdam`,
    offers: generateOffers(5)
  },
  {
    city: `Hamburg`,
    offers: generateOffers(4)
  },
  {
    city: `Dusseldorf`,
    offers: generateOffers(5)
  }
];
