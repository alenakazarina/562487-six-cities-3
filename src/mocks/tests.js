import configureStore from 'redux-mock-store';
import {AuthStatus} from '../reducers/user/user';
import {cityOffers, reviews, appUsers} from './const';

const defaultUser = {
  id: -1,
  email: ``,
  name: ``,
  isPro: false,
  avatarUrl: ``
};

const mockStore = configureStore([]);

export const storeWithAuth = mockStore({
  USER: {
    authStatus: AuthStatus.AUTH,
    user: appUsers[0]
  },
  OFFERS: {
    offers: cityOffers,
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: cityOffers[1],
    activeOffer: cityOffers[2],
    nearOffers: cityOffers.slice(0, 2),
    comments: reviews
  },
  FAVORITES: {
    favorites: cityOffers
  }
});

export const storeWithNoAuth = mockStore({
  USER: {
    authStatus: AuthStatus.NO_AUTH,
    user: defaultUser
  },
  OFFERS: {
    offers: cityOffers,
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: null,
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  FAVORITES: {
    favorites: []
  }
});

export const storeWithNoOffers = mockStore({
  USER: {
    authStatus: AuthStatus.NO_AUTH,
    user: defaultUser
  },
  OFFERS: {
    offers: [],
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: null,
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  FAVORITES: {
    favorites: []
  }
});
