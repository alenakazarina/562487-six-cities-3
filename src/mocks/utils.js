import nanoid from 'nanoid';
import {IMAGE_PATH, OffersData} from './const';

const getRandomNumber = ({min = 0, max = 1}) => {
  return min + Math.floor(max * Math.random());
};

const getRandomFromObject = (obj = {}) => {
  const randomKey = getRandomFromArray(Object.keys(obj));
  return obj[randomKey];
};

const getRandomFromArray = (arr = []) => {
  const randomIndex = getRandomNumber({min: 0, max: arr.length});
  return arr[randomIndex];
};

const getRandomArray = ({items = [], minLength = 0, maxLength = 1}) => {
  const length = getRandomNumber({min: minLength, max: maxLength + 1});
  return new Array(length).fill(``).map(() => {
    const index = getRandomNumber(0, items.length);
    return items[index];
  });
};

const getRandomBoolean = () => {
  return Math.random() > 0.5;
};

const generateId = () => {
  return nanoid();
};

const getImage = (items = []) => {
  const randomIndex = getRandomNumber({min: 0, max: items.length});
  return IMAGE_PATH.concat(items[randomIndex]);
};

const generateImages = () => {
  return OffersData.images.map((image) => IMAGE_PATH.concat(image));
};

const formatMonthYear = (date = new Date()) => {
  return `${date.getMonth() + 1} ${date.getFullYear()}`;
};

export {
  getRandomNumber,
  getRandomArray,
  getRandomBoolean,
  generateId,
  getImage,
  generateImages,
  getRandomFromArray,
  getRandomFromObject,
  formatMonthYear
};
