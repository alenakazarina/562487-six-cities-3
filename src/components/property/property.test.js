import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {storeWithAuth} from '../../mocks/tests';
import Property from './property';
import {cityOffers, appUsers, reviews} from '../../mocks/const';

const mockFn = () => {};

describe(`Property`, () => {
  it(`should render Property correctly`, () => {
    const tree = renderer.create(
        <Provider store={storeWithAuth}>
          <BrowserRouter>
            <Property
              isAuth
              user={appUsers[0]}
              errorStatus={0}
              activeOffer={cityOffers[0]}
              nearOffers={cityOffers.slice(1)}
              reviews={reviews}
              onReviewSubmit={mockFn}
              onOfferPageLoad={mockFn}
            />
          </BrowserRouter>
        </Provider>,
        {createNodeMock: () => document.createElement(`div`)}
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
