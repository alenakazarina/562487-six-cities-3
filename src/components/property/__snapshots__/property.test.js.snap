// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property should render Property correctly 1`] = `
<div
  className="page page--property"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={
                    Object {
                      "backgroundImage": "url(https://htmlacademy-react-3.appspot.com/six-cities/user1_avatar)",
                    }
                  }
                />
                <span
                  className="header__user-name user__name"
                >
                  user@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
            style={
              Object {
                "width": "259px",
              }
            }
          >
            <img
              alt="Interior photo"
              className="property__image"
              src="image1"
            />
          </div>
          <div
            className="property__image-wrapper"
            style={
              Object {
                "width": "259px",
              }
            }
          >
            <img
              alt="Interior photo"
              className="property__image"
              src="image2"
            />
          </div>
          <div
            className="property__image-wrapper"
            style={
              Object {
                "width": "259px",
              }
            }
          >
            <img
              alt="Interior photo"
              className="property__image"
              src="image3"
            />
          </div>
          <div
            className="property__image-wrapper"
            style={
              Object {
                "width": "259px",
              }
            }
          >
            <img
              alt="Interior photo"
              className="property__image"
              src="image4"
            />
          </div>
          <div
            className="property__image-wrapper"
            style={
              Object {
                "width": "259px",
              }
            }
          >
            <img
              alt="Interior photo"
              className="property__image"
              src="image5"
            />
          </div>
          <div
            className="property__image-wrapper"
            style={
              Object {
                "width": "259px",
              }
            }
          >
            <img
              alt="Interior photo"
              className="property__image"
              src="image6"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Amazing and Extremely Central Flat
            </h1>
            <button
              className="property__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height={33}
                width={31}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              5
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3 Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              5 adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              400
            </b>
            <span
              className="property__price-text"
              style={
                Object {
                  "marginLeft": "3px",
                }
              }
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Breakfast
              </li>
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                className="property__inside-item"
              >
                Washer
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
              >
                <img
                  alt="Alena avatar"
                  className="property__avatar user__avatar"
                  height={74}
                  src="..//user1_avatar"
                  width={74}
                />
              </div>
              <span
                className="property__user-name"
              >
                Alena
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                2
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper reviews__avatar-wrapper--pro"
                  >
                    <img
                      alt="Alena avatar"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="/user1_avatar"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Alena
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020:03:04T14:00"
                  >
                    March 2020
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper reviews__avatar-wrapper--pro"
                  >
                    <img
                      alt="Nick avatar"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="/user2_avatar"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Nick
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020:03:03T10:00"
                  >
                    March 2020
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form"
              method="post"
              onSubmit={[Function]}
            >
              <fieldset
                style={
                  Object {
                    "border": "none",
                    "padding": "0",
                  }
                }
              >
                <label
                  className="reviews__label form__label"
                  htmlFor="review"
                >
                  Your review
                </label>
                <div
                  className="reviews__rating-form form__rating"
                >
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="5-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={5}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="5-stars"
                    title="perfect"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="4-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={4}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="4-stars"
                    title="good"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="3-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={3}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="3-stars"
                    title="not bad"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="2-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={2}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="2-stars"
                    title="badly"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    id="1-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={1}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="1-stars"
                    title="terribly"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                </div>
                <textarea
                  className="reviews__textarea form__textarea"
                  id="review"
                  name="review"
                  onChange={[Function]}
                  placeholder="Tell how was your stay, what you like and what can be improved"
                  value=""
                />
              </fieldset>
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a>
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="image1"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    400
                  </b>
                  <span
                    className="place-card__price-text"
                    style={
                      Object {
                        "marginLeft": "3px",
                      }
                    }
                  >
                    / night
                  </span>
                </div>
                <button
                  className="near-places__bookmark-button button place-card__bookmark-button--active"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/1"
                  onClick={[Function]}
                >
                  Amazing and Extremely Central Flat
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a>
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="image3"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    500
                  </b>
                  <span
                    className="place-card__price-text"
                    style={
                      Object {
                        "marginLeft": "3px",
                      }
                    }
                  >
                    / night
                  </span>
                </div>
                <button
                  className="near-places__bookmark-button button place-card__bookmark-button--active"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/2"
                  onClick={[Function]}
                >
                  The house among olive
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                House
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
